### ISSUE TITLE
Add support for the Checks API

### ISSUE THREAD

**Issue by serkankirbas** [github-api/github-api#520]
The Github v3 API now supports a first-class notion of [check runs](https://developer.github.com/v3/checks/runs/) and [check suites](https://developer.github.com/v3/checks/suites/). Rather than binary pass/fail build statuses, integrations can report rich statuses, annotate lines of code with detailed information, and kick off reruns. 

Github checks are becoming the de facto approach for implementing bots and for users who want to develop custom checks. I think supporting Checks API should be a high priority.

**Comment by bitwiseman** [comment]
Agreed. Those would be great!  Do you have time to submit a PR?

**Comment by XiongKezhi** [comment]
Does the work to represent the Check Suite event payload still need any help?  I am trying to implement the Checks API as a Jenkins plugin for a GSoC idea (see [GitHub Checks API for Jenkins Plugins ](https://jenkins.io/projects/gsoc/2020/project-ideas/github-checks/)), and I'm happy to help with this, so it can make my later implementations easier!

**Comment by bitwiseman** [comment]
It looks like we have read access to checks runs and checkrun events, but not suites and write access. 

https://github.com/github-api/github-api/search?q=GHCheckRun&unscoped_q=GHCheckRun

You'll want to look at the links in the description and implement those. I'd suggest starting off with some smaller PRs or open a draft PR so I can give you pointers as you go along.

**Comment by XiongKezhi** [comment]
Thanks, the links are really helpful, I was strange about those class without setters before. I'll start off with a draft PR on CheckSuite and CheckSuiteEvent.

**Comment by bitwiseman** [comment]
Fixed by #723 .

**Comment by XiongKezhi** [comment]
Sorry, I just saw this message. Also, until I saw those builders and  [#724](https://github.com/github-api/github-api/pull/724), I thought our project left other creating works to the users themselves because we have already provided a pluggable Http client. I wonder whether our current ways to implement those CRUD methods are to follow the design of [#724](https://github.com/github-api/github-api/pull/724)? Like for creating a `GHCommit`, we use a `GHCommitBuilder`, so we also need a `GHCheckRunBuilder` or `GHCheckSuiteBuilder` to create them?
